engine/src/flutter/lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -443,6 +444,12 @@ enum EngineSemanticsRole {
   /// A cell in a [table] contains header information for a column.
   columnHeader,
 
+  /// A container that display its children in a list layout.
+  list,
+
+  /// An item in a [list].
```

Comment:
Does `listItem` need to be a direct child of a `list`? We have a similar situation/problem with tables.

Reply:
Cool. Do we want to document and/or enforce the relationship in the node structure somehow?

Reply:
for semanticsnode, yes we have check in framework side. i.e. _semanticsListItem

or do you meant in dom structure?

Reply:
A check in the framework is probably good enough, and it's probably easier to maintain there.